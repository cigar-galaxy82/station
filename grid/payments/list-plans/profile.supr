name = "payments/list-plans"
version = "1.0.0"

"""
List subscription plans
"""

usecase ListPlans safe {
  result [
    {
      """
      ID
      Unique identifier of the subscription plan.
      """
      id string!

      """
      Name
      Name of the subscription plan.
      """
      name string!
      
      """
      Interval
      Frequency of the interval at which the customer is charged.

      Possible values:
      - `day`
      - `week`
      - `month`
      - `year`
      """
      interval enum {
        day, week, month, year
      }

      """
      Price
      Amount charged each `interval` cycle.
      """
      price number!

      """
      Currency code
      Three-letter currency code in which `price` is specified, in ISO 4217 format.
      """
      currency string!
    }
  ]
  

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    result [
      {
        id = "PLAN-89101829211"
        name = "Monthly subscription to Superface milk delivery"
        interval = "month"
        price = 25
        currency = "USD"
      },
      {
        id = "PLAN-89101829212"
        name = "Daily subscription to Superface pizza delivery"
        interval = "day"
        price = 5
        currency = "USD"
      }
    ]
  }
}