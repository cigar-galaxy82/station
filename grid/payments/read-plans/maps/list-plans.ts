/* eslint-disable jest/no-export */

import { describe, expect } from '@jest/globals';
import { SuperfaceTest } from '@superfaceai/testing';

export function listPlansTest(providerName: string): void {
  describe(`payments/list-plans/${providerName}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'payments/list-plans',
        provider: providerName,
        testInstance: expect,
      });
    });

    describe('ListPlans', () => {
      describe('when all inputs are correct', () => {
        it('creates a product', async () => {
          const result = await superface.run({
            useCase: 'ListPlans',
            input: {},
          });
          expect(() => result.unwrap()).not.toThrow();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
}
