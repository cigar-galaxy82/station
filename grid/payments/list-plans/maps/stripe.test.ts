import { listPlansTest } from './list-plans';

listPlansTest('stripe');
