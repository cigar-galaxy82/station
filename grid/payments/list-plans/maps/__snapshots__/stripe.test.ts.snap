// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/list-plans/stripe ListPlans when all inputs are correct creates a product 1`] = `
Array [
  Object {
    "id": "price_1LwodKGkdIyBfLnZ12NX05vs",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwodJGkdIyBfLnZLEliA6wN",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwodJGkdIyBfLnZqwMV2nK2",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoczGkdIyBfLnZ67aLtcI6",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoczGkdIyBfLnZuxilSKsl",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwoczGkdIyBfLnZqgdIYIVM",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoLvGkdIyBfLnZvCH1Lz6M",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwoLtGkdIyBfLnZ1ZcaYPIK",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoLtGkdIyBfLnZdVMTQnwd",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoK6GkdIyBfLnZ6fhhNKn9",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwoK5GkdIyBfLnZ6vtWLAI5",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwoK5GkdIyBfLnZAqh9Bys1",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1Lwn89GkdIyBfLnZP5DRcDMI",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1Lwn87GkdIyBfLnZLSpB5cRs",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1Lwn87GkdIyBfLnZpU6J0kbf",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmxxGkdIyBfLnZS5y2duzL",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwmxuGkdIyBfLnZcX4g4ULA",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmxuGkdIyBfLnZ0GRfu5xc",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmxcGkdIyBfLnZKvQuplsF",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmwmGkdIyBfLnZajkoCvev",
    "name": "SF Product Plan2 Test",
    "status": "active",
  },
  Object {
    "id": "price_1LwmwdGkdIyBfLnZNPq6t75y",
    "name": "SF Product Plan2 Test",
    "status": "active",
  },
  Object {
    "id": "price_1LwmvtGkdIyBfLnZ8BZw1QIF",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmrsGkdIyBfLnZlvOqADeo",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmrEGkdIyBfLnZl6ePt3nA",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmooGkdIyBfLnZLFCA1O4v",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwmomGkdIyBfLnZ15kjcWUH",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmomGkdIyBfLnZEnjuxAKD",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmmEGkdIyBfLnZ3lScdkTt",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1Lwmm9GkdIyBfLnZLAehCfAz",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1Lwmm6GkdIyBfLnZQytQapK9",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
  Object {
    "id": "price_1LwmiIGkdIyBfLnZzKO88ZYe",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmYzGkdIyBfLnZqBSHV74e",
    "name": "Temporary Plan",
    "status": "active",
  },
  Object {
    "id": "price_1LwmOEGkdIyBfLnZhzPCfeyp",
    "name": "A Man, A Plan, A Canal, Panama",
    "status": "active",
  },
]
`;
